includes/gateway/abstract-omise-payment-base-card.php

```diff
@@ -24,7 +24,11 @@ public function charge($order_id, $order)
 		}
 
 		$user = $order->get_user();
-		$omise_customer_id = $this->is_test() ? $user->test_omise_customer_id : $user->live_omise_customer_id;
+		$omise_customer_id = null;
+
```

Comment:
Let's start breaking functions into smaller unit when it makes sense. Here, I see we can move the responsibility of getting customer ID to a different method thus adhering to `separation of concerns` and `single responsibility principle`. Plus, if we had to make any change to get the customer ID in the future then we do not need to touch `charge` function adhering to `open-close principle`.


```
$omise_customer_id = $this->getCustomerId($user)
...
...
private function getCustomerId($user)
{
	if(!empty($user)) {
		return null;
	}

	return $this->is_test() ? $user->test_omise_customer_id : $user->live_omise_customer_id;
}
```

Reply:
Fixed.